### PR TITLE
ROX-16774: byodb stop copies

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

The idea was to stop making copies but when I got into I realized that had to be phased in.  For instance if we go from 3.74 or 4.0 to 4.2 the only way to rollback is if we support central_previous because those releases expect the DB copies to exist.  So this will take a step and not make copies if coming from 4.1.  In ROX-18005 we will phase out the rest of the copy logic when we no longer support rollbacks to Postgres backed database from 4.0 or before.

That made this a little messier than I wanted but I noted many TODOs related to ROX-18005 such that we can track them down and handle accordingly.

Note:  CHANGELOG was updated in #6623 to say that in the future we will not support rollbacks to release 4.0 or prior.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Added some unit tests.  Ran various upgrade scenarios manually.  More testing to come as the rest of ROX-16252 is implemented.

### CODE REVIEW

File: central/globaldb/postgres.go [hunk 1]
```diff
@@ -104,14 +104,6 @@ func InitializePostgres(ctx context.Context) postgres.DB {
 			log.Fatalf("Could not parse postgres config: %v", err)
 		}
 
-		if !pgconfig.IsExternalDatabase() {
```
Comment (dashrews78) [inline central/globaldb/postgres.go:107]: for ACS hosted Posgres, with this PR `central_active` will be created as part of the migrator.  So with the exception of restores, central will not need to do anything with the database name.

File: migrator/postgreshelper/utils.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package postgreshelper
```
Comment (dashrews78) [inline migrator/postgreshelper/utils.go:1]: Moved this from pgAdmin to the migrator to isolate it to where it is used to more easily remove it later.

File: migrator/clone/postgres/db_clone_manager_impl_external_test.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+func (s *PostgresExternalManagerSuite) SetupTest() {
+	ctx := sac.WithAllAccess(context.Background())
+
+	// Create the external database.
+	pgtest.DropDatabase(s.T(), externalDB)
+	pgtest.CreateDatabase(s.T(), externalDB)
+
+	// Update the default configs to use the external database
+	config.GetConfig().CentralDB.External = true
```
Comment (connorgorman) [inline migrator/clone/postgres/db_clone_manager_impl_external_test.go:53]: No need to use Getconfig here and can just use Config

Comment (connorgorman) [reply migrator/clone/postgres/db_clone_manager_impl_external_test.go:53]: Nvm, mistaken. Thought it was a proto

**Review by connorgorman — APPROVED** [review]
This looks good, one thing I think we should do in a follow up is to start running a separate and slimmed down upgrade test with an external DB as there is a lot of if statements that are testing when it is enabled

**Pull request merged**